src/Tests/Microsoft.NET.Sdk.Razor.Tests/AspNetSdkBaselineTest.cs

```diff
@@ -356,8 +386,36 @@ string UpdateAssetVersion(string restorePath, string property)
                     .Should()
                     .BeEquivalentTo(expected.ReferencedProjectsConfiguration.OrderBy(cm => cm.Identity));
                 manifest.DiscoveryPatterns.OrderBy(dp => dp.Name).ShouldBeEquivalentTo(expected.DiscoveryPatterns.OrderBy(dp => dp.Name));
-                manifest.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind)
-                    .ShouldBeEquivalentTo(expected.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind));
+
+                var manifestAssets = manifest.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind);
+                var expectedAssets = expected.Assets.OrderBy(a => a.BasePath).ThenBy(a => a.RelativePath).ThenBy(a => a.AssetKind);
+                
+                manifestAssets.ShouldBeEquivalentTo(expectedAssets, AssetDifferencesDetails(manifestAssets, expectedAssets));
+
+                static string AssetDifferencesDetails(IEnumerable<StaticWebAsset> manifestAssets, IEnumerable<StaticWebAsset> expectedAssets)
+                {
+                    var missingAssets = expectedAssets.Except(manifestAssets);
+                    var unexpectedAssets = manifestAssets.Except(expectedAssets);
+
+                    var differences = new List<string>();
+
+                    if (missingAssets.Any())
+                    {
+                        differences.Add($"The following expected assets weren't found in the manifest {string.Join(", ", missingAssets.Select(a => a.Identity))}.");
+                    }
+
+                    if (unexpectedAssets.Any())
+                    {
+                        differences.Add($"The following additional unexpected assets were found in the manifest {string.Join(", ", unexpectedAssets.Select(a => a.Identity))}.");
+                    }
+
+                    if (differences.Any())
+                    {
+                        differences.Add("If the difference in baselines is expected, please re-generate the baselines using the steps at the top of this file in the source code.");
```

Comment:
This will be in the log, right? Will it be obvious what "this file" refers to?

Reply:
Yep, it should be clear by the stack trace. I can add in a relative path if you think it'd be helpful.